--- bCNC/CNC.py	2020-05-21 10:49:55.615985443 +0200
+++ bCNC/CNC.py	2020-05-21 10:53:00.038238946 +0200
@@ -1039,7 +1039,7 @@ class CNC:
 	#----------------------------------------------------------------------
 	@staticmethod
 	def glinev(g, v, feed=None):
-		pairs = zip("xyz",v)
+		pairs = list(zip("xyz",v))
 		if feed is not None:
 			pairs.append(("f",feed))
 		return CNC.gcode(g, pairs)
@@ -1047,7 +1047,7 @@ class CNC:
 	#----------------------------------------------------------------------
 	@staticmethod
 	def garcv(g, v, ijk):
-		return CNC.gcode(g, zip("xyz",v) + zip("ij",ijk[:2]))
+		return CNC.gcode(g, list(zip("xyz",v)) + list(zip("ij",ijk[:2])))
 
 	#----------------------------------------------------------------------
 	@staticmethod
